src/transformers/modeling_utils.py

```diff
@@ -2549,26 +2549,20 @@ def save_pretrained(
         # Save the config
         if is_main_process:
             if not _hf_peft_config_loaded:
+                # If the model config has set attributes that should be in the generation config, move them there.
```

Comment:
Instead of warning (before this PR) / raising an exception (this PR if L2552-2563 were not added) when `model.config` holds things that should be in `model.generation_config` AND were explicitly set by the user (e.g. `model.config.top_k = 5`), now we automagically move misplaced items ✨ 

Automagically moving things is not usually recommended, but this is a special case: `model.save_pretrained` is often run at the end of training jobs, we really want to avoid expensive crashes